### PR TITLE
ctags detection should be only called when needed

### PR DESCRIPTION
**Pull request by vigsterkr**
but then it's required

### CODE REVIEW

**Comment by vigsterkr** [comment]
@gf712 right?

File: CMakeLists.txt [hunk 1]
```diff
@@ -555,24 +555,26 @@ IF(ENABLE_TESTING AND NOT BUILD_DASHBOARD_REPORTS)
...
-    --exclude=rxcpp
-    --exclude=interfaces
-    --exclude=Eigen3
-    --exclude=external
-    --languages=c++
-    -R ${CMAKE_SOURCE_DIR})
-
-ADD_CUSTOM_TARGET(ctags DEPENDS ${CTAGS_FILE})
+IF (DEFINED CTAGS_FILE)
+    FIND_PACKAGE(Ctags REQUIRED)
```
Comment (gf712) [inline CMakeLists.txt]: why required? if it doesn't find it then the command will not run right?

**Comment by Hephaestus12** [comment]
Is it possible to set this CTAGS_FILE manually while running cmake?

**Comment by vigsterkr** [comment]
this should fix #5003

**Pull request merged**